sklearn/decomposition/pca.py

```diff
@@ -448,7 +455,8 @@ def _fit_full(self, X, n_components):
         # Postprocess the number of components required
         if n_components == 'mle':
             n_components = \
-                _infer_dimension_(explained_variance_, n_samples, n_features)
+                _infer_dimension_(explained_variance_, n_samples,
+                                  n_features) + 1
```

Comment:
you cannot change the behavior in any case like this. Was is it a bug? when I complained before about n_components=0 I was surprised but was the behavior expected for MLE option?

Reply:
I am not sure. But this off by one problem is discussed here. https://github.com/scikit-learn/scikit-learn/pull/4827#discussion_r31869858

Reply:
I don't think there was an off-by-one error here formerly, the argmax seems to correspond correctly to the assessed rank. But if rank=0 is a bad answer, then we don't need to assess rank=0... If so, that's a separate bug, but this +1 should be removed, IMO.